### DISCUSSION TITLE
Specifications for how to integrate services in AI models

### DISCUSSION THREAD

**Discussion by PengXing** [ggml-org/llama.cpp#810] [General]
Please join discussion. 

Here is a repository I created for specs, for now it's empty. Discussion is welcome. https://github.com/openservices4ai/spec

**What is an AI-integrated sevice** 

As AI models trained on a fixed set of data and cannot learn something new in real-time while serving, services integrated in AI models help provide real-time information and services. For example, a chatbot model may be integrated with a todo list service to help users add/remove todos, or create a timer. 

**Why an unified specification for AI-integrated services matters**

First, it can help improve compatibility between different AI platforms, making it easier for developers to integrate various plugins into their AIs. This can save time and resources for developers who would otherwise have to manually modify each plugin to work with their chatbot.

Second, an unified specification can help ensure that services adhere to certain standards for functionality, security, and user privacy. This can help promote trust and confidence among users who interact with chatbots using these services.

Finally, it can also help promote innovation and collaboration among developers working on AI chatbots. By providing a common framework for plugin development, developers can more easily share their work and collaborate on new features and functionalities, ultimately leading to better and more sophisticated chatbots for users.

**Comment by BarfingLemurs** [comment]
(this post should be here): https://github.com/ggerganov/llama.cpp/discussions/categories/general

**Reply by prusnak** [reply]
converted the issue to discussion post
